### DISCUSSION TITLE
Using Optuna to find the best imputer, ran into a strat problem beforehand because of Nan values...

### DISCUSSION THREAD

**Discussion by dec1costello** [optuna/optuna#5522] [Q&A]
Hello I am attempting to hyperparameter tune different imputation methods for my model. Before hand, I want to stratify my data, but is this possible when the rows I strat by contain Nans?

Here is my attempt to strat:
'''
X = result_df[feature_cols]
y = result_df['strokes_to_hole_out']

#Extract the columns for stratification
stratify_cols = ['from_location_scorer','from_location_laser']
stratify_data = result_df[stratify_cols]

#Split the data, using 'stratify_data' for stratification
X_train, X_valid, y_train, y_valid = train_test_split(X, y, test_size=0.2, random_state=42, stratify=stratify_data)

'''
but come training, here is my error: Trial failed with exception: Found unknown categories ['blue'] in column 9 during transform

Thanks in advance!!

**Comment by nzw0301** [comment]
I'm afraid that I think this is not related to optuna, so asking your question on different github repo or platform is more likely to get response. Maybe scikit-learn because it looks [`sklearn.model_selection.train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html)?
